TF_Tooling_Web/src/TouchFree.ts

```diff
@@ -0,0 +1,29 @@
+import { ConnectionManager } from "./Connection/ConnectionManager";
+import { SVGCursor, SVGCursorParams } from "./Cursors/SvgCursor";
+import { TouchlessCursor } from "./Cursors/TouchlessCursor";
+import { WebInputController } from "./InputControllers/WebInputController";
+
+var InputController: WebInputController;
+export var CurrentCursor: TouchlessCursor;
+
+export function Init(_cursor: TouchlessCursor): void;
+export function Init(_svgParams?: SVGCursorParams): void;
+export function Init(_cursorOrParams?: SVGCursorParams | TouchlessCursor): void {
```

Comment:
I feel like we could just have an object that handles these two cases e.g.
`{ cursor: TouchlessCursor, svgCursorParams: SVGCursorParams, }`
And as long as we make it clear that you can create this shape object then it would be clearer and be more extensible in future.

Reply:
I agree with Gabe - there should be an optional options interface you can pass in the init function?

export function Init(options?: TouchFreeInitOptions)

interface TouchFreeInitOptions {
     cursor?: TouchlessCursor
}

Instead of passing svgCursor params could the user not create an SVGCursor (using the params) then pass it in via the TouchFreeInitOptions?

Reply:
See my suggestion here https://github.com/ultraleap/TouchFree/pull/381/commits/9425a08f4a0fd9ca078e14e8be4846b1a895d307#r1012056817

Reply:
I don't think the SVGCursorParams are needed? Can't they make a new SVG cursor then pass it in? Or is that a weird / hidden workflow

Reply:
I'd designed for three approaches to the Init:

1. Just make it work, I don't care for the details
2. This ^ but I wanna tweak something
3. I want full control, use this thing I made

2 & 3 could both be fulfilled by just the Cursor passing, only difference is it requires a user to 
understand that the SVG cursor is the default, & then the step of them making said cursor beforehand.

All this said, I'm starting to fear that all of this is actually redundant; I'm not sure the case of passing in a cursor *will ever work*. Cursors won't be able to initialise successfully without the ConnectionManager already being instanced... so you need some of the function to have run before you instace the thing you're passing _into_ that function. 
We could replace this with a factory function you provide maybe? As a "once you've initialised the connection, use this function to make a Cursor"?

Reply:
Maybe then we only support 1? And if you want a non-default cursor you have to do it the long way?

Reply:
Maybe we support 1 & 3 but we have two seperate functions:

`InitAll` does what `Init()` currently does; gets you a Connection, Inputs & a Cursor
`InitBase` only initialises the bits you don't need to think about (Connection & Inputs) and leaves you to manage your cursor(s) as you please?

Reply:
Alternatively maybe it's one function with a bool dictating whether it makes a cursor for you, which defaults to true?

Reply:
^ personally I like that option the most (or could make the bool inside the InitParams interface and have DefaultInitParms - but that's only useful if we envision adding lots more options)

Reply:
The change I described in my previous comment is now implemented 👍